lib/FundDistribution/FundDistributionFields/FundDistributionFieldsFinalContainer.js

```diff
@@ -36,26 +39,51 @@ const FundDistributionFieldsFinalContainer = ({
   const [expenseClassesByFundId, setExpenseClassesByFundId] = useState({});
 
   useEffect(() => {
-    const expenseClassesFundsToFetch = fundDistribution?.filter(({ fundId }) => (
-      fundId && !expenseClassesByFundId[fundId]
-    ));
-
-    if (expenseClassesFundsToFetch?.length) {
-      expenseClassesFundsToFetch.forEach(({ fundId }) => fetchExpenseClasses(
-        fundId, expenseClassesByFundId, setExpenseClassesByFundId, mutator.fundExpenseClasses,
-      ));
+    const uniqueFundDistribution = uniqBy(fundDistribution, 'fundId');
+
+    uniqueFundDistribution.forEach(({ fundId }) => fetchExpenseClasses({
+      fundId,
+      fiscalYearId,
+      setExpenseClassesByFundId,
+      mutator: mutator.fundExpenseClasses,
+    }));
+  }, [fundDistribution, fiscalYearId, mutator.fundExpenseClasses]);
+
+  const resetExpenseClasses = useCallback(() => {
+    formValues?.adjustments?.forEach(({ fundDistributions }, index) => {
```

Comment:
The last comment from the prev PR https://github.com/folio-org/stripes-acq-components/pull/698#discussion_r1219414120

Reply:
What do you suggest to reset data?

Reply:
What I see from the request we are passing the same data as the example and it requires resetting data in the specified order. Here is the POST data to create an invoice with an expanse class id. If you have any suggestions please let me know. @NikitaSedyx @usavkov-epam 
```
{
    "chkSubscriptionOverlap": true,
    "currency": "USD",
    "source": "User",
    "adjustments": [
        {
            "type": "Amount",
            "description": "111",
            "value": 1,
            "relationToTotal": "In addition to",
            "prorate": "Not prorated",
            "fundDistributions": [
                {
                    "distributionType": "percentage",
                    "value": 100,
                    "fundId": "7fbd5d84-62d1-44c6-9c45-6cb173998bbd",
                    "code": "AFRICAHIST",
                    "encumbrance": null,
                    "expenseClassId": "1bcc3247-99bf-4dca-9b0f-7bc51a2998c2"
                }
            ]
        },
        {
            "type": "Amount",
            "description": "222",
            "value": 1,
            "relationToTotal": "In addition to",
            "prorate": "Not prorated",
            "fundDistributions": [
                {
                    "distributionType": "percentage",
                    "value": 100,
                    "fundId": "7fbd5d84-62d1-44c6-9c45-6cb173998bbd",
                    "code": "AFRICAHIST",
                    "encumbrance": null,
                    "expenseClassId": "1bcc3247-99bf-4dca-9b0f-7bc51a2998c2"
                }
            ]
        }
    ],
    "batchGroupId": "cd592659-77aa-4eb3-ac34-c9a4657bb20f",
    "status": "Open",
    "exportToAccounting": true,
    "vendorId": "e0fb5df2-cdf1-11e8-a8d5-f2801f1b9fd1",
    "invoiceDate": "2023-06-06",
    "fiscalYearId": "0bb73cfd-9251-44ce-a7de-3ef16875ba6a",
    "accountingCode": "G64758-74834",
    "paymentMethod": "Credit Card",
    "vendorInvoiceNo": "00000",
    "accountNo": "G64758-74834",
    "id": "93ae4405-d5db-43aa-a67a-0785a7a75379"
}
```

Reply:
the proposal is to implement all connected to invoices in ui-invoices